Group level visits

hardiesoft opened the pull request:
Let's get main synced up with the latest release

mjs commented on public/javascripts/viewRecording.js:
```diff
@@ -2,6 +2,7 @@ var recordingsApiUrl = api + '/api/v1/recordings'
 var recording = null;
 
 document.onkeypress = function (e) {
+  if (document.activeElement.tagName != 'BODY') { return; }
```
will that tag name always be upper case? (i.e. should we be forcing it to upper case before checking?)
will the active element *always* be the body if an input field isn't selected?

CameronRP replied:
I would assume that the tag name will always be upper case but can change that.
I just chose to use 'BODY' as I'm not too sure on what all the input tag names might be in the future. It looks like the only ones that are an input right now are 'INPUT' and 'SELECT'. Should I just change it to ignore it when this is the case?

mjs replied:
I don't know enough about how `activeElement` works in all browsers to be sure. I guess leave it as is and see how it goes. At worst it'll disable the shortcuts more than it should.

Forcing the tag name to upper or lowercase before comparing is probably a good idea though.

Commit:
add reset methods and fields

Commit:
add files

hardiesoft committed:
Allow duplicate recordings to return 200 response.

hardiesoft committed:
Allow duplicate recordings to return 200 response (2)

hardiesoft committed:
Allow duplicate recordings to return 200 response (2)

hardiesoft committed:
Allow duplicate recordings to return 200 response (2)

hardiesoft committed:
Add undelete endpoint

hardiesoft committed:
Linting

hardiesoft committed:
Linting

hardiesoft committed:
Linting (2)

Commit:
added new passowwrd

Commit:
tidy up

Commit:
tidied up

Commit:
add api docs

Commit:
Merge remote-tracking branch 'origin/main' into reset-password

hardiesoft committed:
Undelete UI

hardiesoft committed:
Merge branch 'main' of https://github.com/TheCacophonyProject/cacophony-web into undelete

hardiesoft committed:
Fix resolution of apiInterface, add some typing to front-end api

hardiesoft committed:
Correct undelete endpoint

Commit:
dont import interface

Commit:
Merge branch 'reset-password' of https://github.com/TheCacophonyProject/cacophony-web into reset-password

Commit:
updated formatting

Commit:
update email template

Commit:
add heartbeat endpoint

Commit:
update

Commit:
update

Commit:
update

Commit:
update

Commit:
lint fix

Commit:
await

Commit:
Merge remote-tracking branch 'origin/main' into heart-beat

hardiesoft committed:
WIP schedules

hardiesoft committed:
Fix undelete api endpoint order

Commit:
catch exception from jwt and pass it on

Commit:
update report to use heartbeat

Commit:
update

hardiesoft committed:
WIP Audiobait scheduler UI.

hardiesoft committed:
Audiobait scheduler UI.

hardiesoft committed:
Merge branch 'heart-beat' of https://github.com/TheCacophonyProject/cacophony-web into audio-bait-schedules

hardiesoft committed:
Merge branch 'reset-password' of https://github.com/TheCacophonyProject/cacophony-web into audio-bait-schedules

 Conflicts:
	browse/src/router/index.ts

hardiesoft committed:
Fix schema

hardiesoft committed:
Expose heartbeat health to UI.

hardiesoft committed:
Fixup some API issues

hardiesoft committed:
Remove log

hardiesoft committed:
Api fix

Commit:
dont show state until data

Commit:
fix autofill state

Commit:
lint

hardiesoft committed:
Improve stations query handling in query view.  Dedupe devices and stations with name-collisions by group.

hardiesoft committed:
Merge branch 'reset-password' of https://github.com/TheCacophonyProject/cacophony-web into next-rc

hardiesoft committed:
Fix tests for stations and tagging

Commit:
fix bad query

Commit:
Merge remote-tracking branch 'origin/heart-beat' into next-rc

hardiesoft committed:
Fixes to SelectDevice, add get station by id api.

hardiesoft committed:
Merge branch 'next-rc' of https://github.com/TheCacophonyProject/cacophony-web into next-rc

hardiesoft committed:
Fix get station by id

hardiesoft committed:
Fix get station by id (2)

hardiesoft committed:
Update api docs

hardiesoft committed:
Update tests for stations

hardiesoft committed:
Trim off schedule seconds.

Commit:
copy config instead of using different config

Commit:
configure admin emails in config

Commit:
Merge remote-tracking branch 'origin/heart-beat' into next-rc

Commit:
Merge branch 'next-rc' of https://github.com/TheCacophonyProject/cacophony-web into next-rc

hardiesoft committed:
Make it clear that wait times are all in seconds in scheduler UI.

hardiesoft committed:
Merge remote-tracking branch 'origin/next-rc' into next-rc

hardiesoft committed:
Fix super-user check for showing track history.

gw7nvw committed:
initial version - user tests

gw7nvw committed:
initial version - user tests

gw7nvw committed:
All user tests expect pasword reset

hardiesoft committed:
Merge in next-rc changes

hardiesoft committed:
Merge in next-rc test changes

hardiesoft committed:
Merge in next-rc test changes

hardiesoft committed:
Merge in next-rc test changes (3)

hardiesoft committed:
Merge in next-rc test changes (4)

gw7nvw committed:
fixed test user email address

hardiesoft committed:
Merge branch 'add_user_tests' of https://github.com/TheCacophonyProject/cacophony-web into next-rc2

 Conflicts:
	api/api/V1/Group.ts
	api/api/V1/User.ts
	api/migrations/20211117020340-heartbeat.js
	api/package-lock.json
	integration-tests/cypress/commands/api/device.d.ts
	integration-tests/cypress/commands/api/device.ts
	integration-tests/cypress/commands/api/user.d.ts
	integration-tests/cypress/commands/api/user.ts
	integration-tests/cypress/commands/server.ts
	integration-tests/cypress/integration/api/recordings/recording_parameters.ts
	integration-tests/cypress/integration/api/tracks/track_tag_replace.ts
	integration-tests/cypress/integration/api/users-admin/admin_global_user.ts
	integration-tests/cypress/integration/api/users-admin/end_user_agreement.ts
	integration-tests/cypress/integration/api/users-admin/user_add_get.ts
	integration-tests/cypress/integration/api/users-admin/user_list.ts
	integration-tests/cypress/integration/api/users-admin/user_update.ts

hardiesoft committed:
Merge in next-rc test changes (5)

hardiesoft committed:
Merge in next-rc test changes (6)

hardiesoft committed:
Merge in next-rc test changes (7)

hardiesoft committed:
Fix permissions issues with get user api

hardiesoft committed:
Try to fix issue with power-tagger tracks

hardiesoft committed:
Try to fix issue with power-tagger tracks (2)

hardiesoft committed:
Try to fix issue with power-tagger tracks (3)

hardiesoft committed:
Try to fix issue with power-tagger tracks (3)

hardiesoft committed:
Try to fix issue with power-tagger tracks (4)

hardiesoft committed:
Adding group-level aggregated device visits support.

hardiesoft committed:
Use nginx proxy_pass for api calls, so that we can avoid costly CORS requests in browse, speeding up the site.

hardiesoft committed:
Flip lat/lng in database to correct positions for PostGIS functions

hardiesoft committed:
Linting

hardiesoft committed:
Fix lat lng order from api

hardiesoft committed:
Linting

hardiesoft committed:
Fix coordinates order in tests.

hardiesoft committed:
Adding consistent group/device/station links as components.
Add audio button player for previewing audiobait files.

hardiesoft committed:
Unload audio players

hardiesoft committed:
Add a progress ring to audio button player.

hardiesoft committed:
Linting

hardiesoft committed:
- Add small link to location modal in mobile breakpoint on recording summary
- Add map header to device page
- Add device locations map to devices page
- Fix scroll bug when searching on an already scrolled down page.
- Remove top level visits from nav

hardiesoft committed:
- Fixes around visits date display

hardiesoft committed:
- Fix CSS issue with device with no location

Commit:
run tracks query seperate, will speed up query

hardiesoft committed:
- Wrap bootstrap tabs component in a new component to allow it to switch to a column view at mobile breakpoints.

hardiesoft committed:
Fix spacing for "Export Visits" button

hardiesoft committed:
Handle deleted recordings in visits

hardiesoft committed:
Resolve ambigous retired stations urls correctly.

hardiesoft committed:
Resolve ambiguous retired stations urls correctly (2).

hardiesoft committed:
Fix issue with recordings station matching not using only non-retired stations.

hardiesoft committed:
Merge branch 'fix-query' of https://github.com/TheCacophonyProject/cacophony-web into group-level-visits

hardiesoft committed:
Correct typo of seperate->separate

hardiesoft committed:
Fix z-index on user-profile dropdown.

hardiesoft committed:
Lint

hardiesoft committed:
De-scope styles for top-level nav

hardiesoft committed:
Restore logo

hardiesoft committed:
CSS Padding fix
